uro98/assignment4/CountIslands.java

```diff
@@ -0,0 +1,57 @@
+
+public class CountIslands {
+
+  /**
+   * Returns the number of islands in a given map.
+   * @param rowCount  the number of rows in the map
+   * @param colCount  the number of columns in the map
+   * @param map       the boolean map where true is land and false is water
+   * @return the number of islands in integer
+   */
+  public static int countIslands(int rowCount, int colCount, boolean[][] map) {
+    int count = 0;
+
+    for (int y = 0; y < colCount; y++) {
+      for (int x = 0; x < rowCount; x++) {
+        if (findIsland(y, x, rowCount, colCount, map)) {
+          count++;
+        }
+      }
+    }
+
+    return count;
+  }
+
+  private static boolean findIsland(int y, int x, int rowCount, int colCount, boolean[][] map) {
+
+    if (isOutOfBounds(y, x, rowCount, colCount)) {
+      return false;
+    }
+
+    if (map[y][x] == true) {
+      // Disable the land as it has been found
+      map[y][x] = false;
+      
+      // The surrounding land is part of the same island
+      disableSurroundingLand(y, x, rowCount, colCount, map);
```

Comment:
Name is a bit confusing. Something like `visitSurroundingTiles` or something similar would be a bit better.